aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -65,4 +134,112 @@ _bincount_cpu(const Tensor& self, const Tensor& weights, int64_t minlength) {
+  // Add support for weights broadcastable to input
+  bool has_weights = weights.defined();
+  Tensor flattened_weights;
+  if (has_weights) {
+    TORCH_CHECK(
+        weights.sizes() == self.sizes(),
+        "histogram only supports input and weights of the same shape");
+    flattened_weights = weights.flatten(0).contiguous();
+  }
+
+
+  // If range is not defined, compute min and max from the values in the Tensor
+  Scalar min;
+  Scalar max;
+  if (range.has_value()) {
+    min = range.value()[0];
+    max = range.value()[1];
+  } else {
+    min = self.min().item();
+    max = self.max().item();
+  }
+
+  Tensor hist = AT_DISPATCH_ALL_TYPES(
+      self.scalar_type(), "histogram_cpu_uniformbins", [&] {
+        const auto scalar = weights.scalar_type();
+        if (scalar == ScalarType::Undefined || scalar == ScalarType::Float)
+          return _histogram_cpu_template_uniformbins<scalar_t, float>(
+              self.flatten(0).contiguous(),
+              bins,
+              flattened_weights,
+              min.to<scalar_t>(),
+              max.to<scalar_t>(),
+              density);
+        return _histogram_cpu_template_uniformbins<scalar_t, double>(
+            self.flatten(0).contiguous(),
+            bins,
+            flattened_weights.to(kDouble),
+            min.to<scalar_t>(),
+            max.to<scalar_t>(),
+            density);
+      });
+
+  if (density) { // Compute the density
+    double minval = min.to<double>();
+    double maxval = max.to<double>();
+    if (minval == maxval) {
+      minval -= 1;
+      maxval += 1;
+    }
+    hist = hist.to(kDouble);
+    hist *= bins / (maxval - minval) / hist.sum();
+  }
+
+  Tensor edges;
+  if (self.scalar_type() == kFloat) {
+    edges = at::linspace(min, max, bins + 1, self.options());
+  } else {
+    edges = at::linspace(min, max, bins + 1, self.options().dtype(kDouble));
+  }
+
+  return std::make_tuple(hist, edges);
+}
+
+std::tuple<Tensor, Tensor> histogram(
+    const Tensor& self,
+    const Tensor& bins,
+    const Tensor& weights,
+    bool density) {
+
+  TORCH_CHECK(bins.dim() == 1, "bins must be 1d, when a tensor");
+  TORCH_CHECK(
+      at::all(bins.slice(0, 1, bins.numel()) >= bins.slice(0, 0, -1))
+          .item<bool>(),
+      "bins must increase monotonically");
+  Tensor flattened_weights;
+  if (weights.defined()) {
+    TORCH_CHECK(
+        weights.sizes() == self.sizes(),
+        "histogram only supports input and weights of the same shape");
+    flattened_weights = weights.flatten(0).contiguous();
+  }
+  int64_t nbins = bins.size(0) - 1;
+  Tensor index = searchsorted(bins, self, false, true);
+  index.clamp_(0, nbins + 2);
+  index = index.where(self != bins[-1], index - 1);
+  Tensor hist = bincount(index.flatten(0), flattened_weights, nbins + 2);
+
+  hist = hist.slice(0, 1, -1);
+
+  if (density) { // Compute the density
+    hist = hist.to(kDouble);
+    hist /= hist.sum() *
```

Comment:
It's interesting how different this computation is than the previous density computation. I would expect them to appear more similar.